app/validation/validators.py

```diff
@@ -167,3 +150,47 @@ def __call__(self, form, from_field, to_field):
             raise validators.ValidationError(self.messages['INVALID_DATE_RANGE_TO_FROM_SAME'])
         elif from_date > to_date:
             raise validators.ValidationError(self.messages['INVALID_DATE_RANGE_TO_BEFORE_FROM'])
+
+
+class FormResponseRequired(object):
+    field_flags = ('required', )
+
+    def __init__(self, message=None):
+        if not message:
+            message = error_messages['MANDATORY']
+        self.message = message
+
+    def __call__(self, form, field):
+
+        if hasattr(form, 'day'):
```

Comment:
I think this would be more concise if it iterated over all the fields on the current form and checked each had data. The default iterator for a form should allow this.

    form_has_data = False

    for field in form:
        form_has_data |= bool(field.data)

    if not form_has_data:
        raise validators.StopValidation(self.message)
            

That would mean we could just apply to any form without caring about it's structure.